sdk/core/core-amqp/src/errors.ts

```diff
@@ -598,14 +615,16 @@ export function translate(err: AmqpError | Error): MessagingError {
     // translate
     const condition = (err as any).code;
     const description = (err as Error).message;
-    error = new MessagingError(description);
+    const error = new MessagingError(description, err);
     if ((err as any).stack) error.stack = (err as any).stack;
+    let errorType = "SystemError";
     if (condition) {
-      const amqpErrorCondition = SystemErrorConditionMapper[condition as keyof typeof SystemErrorConditionMapper];
-      error.name = ConditionErrorNameMapper[amqpErrorCondition as keyof typeof ConditionErrorNameMapper];
+      const amqpErrorCondition =
```

Comment:
Idle speculation: if this cast is valid, then perhaps `code` could be typed as `keyof typeof SystemErrorConditionMapper` to begin with?